src/tree_cmd.cpp

```diff
@@ -799,6 +837,11 @@ static void TileLoop_Trees(TileIndex tile)
 						}
 						break;
 				}
+
+				/* When using improved trees, when a "alone" tree is dead, a new one is planted immediately. */
+				if (_settings_game.game_creation.tree_placer == TP_IMPROVED && !IsNearbyForest(tile)) {
+					PlantTreesOnTile(tile, treetype, 0, 0);
+				}
```

Comment:
Maybe this case should be tested before converting the tree to a clear tile, to avoid unnecessary work.